tests/Utils/BuildSchemaTest.php

```diff
@@ -713,6 +732,222 @@ enum: MyEnum
         self::assertEquals($rootFields['field2']->deprecationReason, 'Because I said so');
     }
 
+    /**
+     * it('Correctly extend scalar type').
+     */
+    public function testCorrectlyExtendScalarType(): void
+    {
+        $scalarSDL = <<<'GRAPHQL'
+            scalar SomeScalar
+            
+            extend scalar SomeScalar @foo
+            
+            extend scalar SomeScalar @bar
+            
+            GRAPHQL;
+
+        $schema = BuildSchema::build("
+            ${scalarSDL}
+            directive @foo on SCALAR
+            directive @bar on SCALAR
+        ");
+
+        /** @var ScalarType $someScalar */
+        $someScalar = $schema->getType('SomeScalar');
```

Comment:
```suggestion
        $someScalar = $schema->getType('SomeScalar');
        assert($someScalar instanceof ScalarType);
```

https://github.com/webonyx/graphql-php/blob/master/CONTRIBUTING.md#type-assertions